pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -327,32 +328,98 @@ func newDeploymentSettingsConfigureCmd() *cobra.Command {
 	return cmd
 }
 
+func newRepoLookup(repoRoot string) (repoLookup, error) {
+	if repoRoot != "" {
+
+		repo, err := git.PlainOpen(repoRoot)
+		if err != nil {
+			return nil, err
+		}
+
+		h, err := repo.Head()
+		if err != nil {
+			return nil, err
+		}
+
+		return &repoLookupImpl{
+			RepoRoot: repoRoot,
+			Repo:     repo,
+			Head:     h,
+		}, nil
+	}
+	return &noRepoLookupImpl{}, nil
+}
+
+type repoLookup interface {
+	GetRootDirectory(wd string) (string, error)
+	GetBranchName() string
+	RemoteURL() (string, error)
+}
+type repoLookupImpl struct {
+	RepoRoot string
+	Repo     *git.Repository
+	Head     *plumbing.Reference
+}
+
+func (r *repoLookupImpl) GetRootDirectory(wd string) (string, error) {
+	return filepath.Rel(r.RepoRoot, wd)
+}
+
+func (r *repoLookupImpl) GetBranchName() string {
+	return r.Head.Name().String()
+}
+
+func (r *repoLookupImpl) RemoteURL() (string, error) {
+	return gitutil.GetGitRemoteURL(r.Repo, "origin")
+}
+
+type noRepoLookupImpl struct{}
+
+func (r *noRepoLookupImpl) GetRootDirectory(wd string) (string, error) {
+	return ".", nil
+}
+
+func (r *noRepoLookupImpl) GetBranchName() string {
+	return ""
+}
+
+func (r *noRepoLookupImpl) RemoteURL() (string, error) {
+	return "", nil
+}
+
 func configureGit(ctx context.Context, displayOpts *display.Options, be backend.Backend,
 	s backend.Stack, sd *workspace.ProjectStackDeployment, gitSSHPrivateKeyPath string,
 ) error {
+	if sd.DeploymentSettings.SourceContext == nil {
+		sd.DeploymentSettings.SourceContext = &apitype.SourceContext{}
+	}
+	if sd.DeploymentSettings.SourceContext.Git == nil {
+		sd.DeploymentSettings.SourceContext.Git = &apitype.SourceContextGit{}
+	}
+
 	wd, err := os.Getwd()
 	if err != nil {
 		return err
 	}
 
 	repoRoot, err := gitutil.DetectGitRootDirectory(wd)
-	if err != nil {
-		return err
+	if err != nil && !errors.Is(err, gitutil.ErrNoGitRepo) {
+		return fmt.Errorf("Could not determine the git root directory: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("could not determine the git root directory: %w", err)
```
nit: error messages usually don't start with a capital letter